app/assess/models/ui/applicants_response.py

```diff
@@ -312,6 +412,36 @@ def _flatten_field_ids(field_id):
 def _convert_non_number_grouped_fields(
     response: Iterable[dict],
 ) -> tuple[list[dict], set[str]]:
+    """
+    Convert non-number grouped fields in a response into a list of individual text items.
+
+    This function takes a response, which is an iterable of dictionaries representing data fields, and returns a tuple containing
+    a list of dictionaries representing individual text items and a set of field IDs that have been processed. The dictionaries
+    in the list contain data for a single text field and have the following keys:
+        question: a string representing the question for the field
+        field_type: a string representing the type of field, if applicable
+        field_id: a string or list of strings representing the field ID(s)
+        answer: a string representing the answer for the field
+        presentation_type: a string set to "text"
+    The input response should be an iterable of dictionaries with the following keys:
+        presentation_type: a string representing the type of data field, which should be "grouped_fields" for the fields that
+        this function processes
+        field_type: a string representing the type of field, which should not be "numberField" for the fields that this
+        function processes
+        question: a list of strings representing the questions for the fields
+        field_id: a list of strings or lists of strings representing the field IDs for the fields
+        answer: a list of tuples, where each tuple contains a string representing the question for the field and a string
+        representing the answer for the field
+        This function processes the fields in the input response that have a presentation_type of "grouped_fields" and a
+        field_type other than "numberField". It converts each field into a dictionary representing a single text field and
+        appends it to the list of text items. The field IDs for the processed fields are added to the set of processed field
+        IDs.
```

Comment:
Black doesn't wrap these lines automatically so have to do it manually, have tried to fix it up a bit

Flake8 also just warns and doesn't auto fix so that's a tad annoying